plugins/catalog-backend-module-gerrit/src/providers/GerritEntityProvider.ts

```diff
@@ -0,0 +1,165 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TaskRunner } from '@backstage/backend-tasks';
+import { Config } from '@backstage/config';
+import { InputError } from '@backstage/errors';
+import {
+  EntityProvider,
+  EntityProviderConnection,
+  LocationSpec,
+  locationSpecToLocationEntity,
+} from '@backstage/plugin-catalog-backend';
+import fetch, { Response } from 'node-fetch';
+import {
+  GerritIntegration,
+  getGerritProjectsApiUrl,
+  getGerritRequestOptions,
+  parseGerritJsonResponse,
+  ScmIntegrations,
+} from '@backstage/integration';
+import * as uuid from 'uuid';
+import { Logger } from 'winston';
+
+import { readGerritConfigs } from './config';
+import { GerritProjectQueryResult, GerritProviderConfig } from './types';
+
+/** @public */
+export class GerritEntityProvider implements EntityProvider {
+  private readonly config: GerritProviderConfig;
+  private readonly integration: GerritIntegration;
+  private readonly logger: Logger;
+  private readonly scheduleFn: () => Promise<void>;
+  private connection?: EntityProviderConnection;
+
+  static fromConfig(
+    configRoot: Config,
+    options: {
+      logger: Logger;
+      schedule: TaskRunner;
+    },
+  ): GerritEntityProvider[] {
+    const providerConfigs = readGerritConfigs(configRoot);
+    const integrations = ScmIntegrations.fromConfig(configRoot).gerrit;
+    const providers: GerritEntityProvider[] = [];
+
+    providerConfigs.forEach(providerConfig => {
+      const integration = integrations.byHost(providerConfig.host);
+      if (!integration) {
+        throw new InputError(
+          `No gerrit integration found that matches host ${providerConfig.host}`,
+        );
+      }
+      providers.push(
+        new GerritEntityProvider(
+          providerConfig,
+          integration,
+          options.logger,
+          options.schedule,
+        ),
+      );
+    });
+    return providers;
+  }
+
+  constructor(
+    config: GerritProviderConfig,
+    integration: GerritIntegration,
+    logger: Logger,
+    schedule: TaskRunner,
+  ) {
+    this.config = config;
+    this.integration = integration;
+    this.logger = logger.child({
+      target: this.getProviderName(),
+    });
+    this.scheduleFn = this.createScheduleFn(schedule);
+  }
+
+  getProviderName(): string {
+    return `gerrit-provider:${this.config.id}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    await this.scheduleFn?.();
```

Comment:
Can be changed to `scheduleFn()` as it will always be set. See also a more recent state of #10480